common/tools/dev-tool/src/util/browserRelayServer.ts

```diff
@@ -47,7 +54,15 @@ function buildServer(app: Express) {
   app.put("/credential", (req, res) => {
     const id = randomUUID();
     try {
-      const cred = new DefaultAzureCredential(req.body);
+      const cred = new ChainedTokenCredential(
+        new AzurePowerShellCredential(req.body),
+        new AzureCliCredential(req.body),
+        // Keep Environment Credential for packages that have not migrated to Federated Authentication
+        // See the migration guide for more information
+        // https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/1080/Secret-auth-migration
+        new EnvironmentCredential(req.body),
+        new AzureDeveloperCliCredential(req.body),
```

Comment:
I'd probably recommend adding DevCli Cred before the EnvironmentCred

Reply:
I'm curious why Dev Cli Cred should come before the Environment Cred? I'm following the listed order in DAC when adding Dev Cli.

Reply:
The only reason I suggested to change up the order was because we are moving away from ClientSecret/ ClientCert credentials, so they should use EnvironmentCredential as a last resort. In my opinion, Azure Powershell, AzureCLI and AxureDeveloperCli all 3 fall in the same category -  just for different dev tools.

Reply:
That makes sense to me. Addressed in the latest commit!

Reply:
@KarishmaGhiya @maorleger so is this something we would want to consider changing in DefaultAzureCredential as well then? I feel like this reasoning does not just apply to the test credential (or, at the very least, this reasoning doesn't feel like a justification for the test credential and DefaultAzureCredential _differing_).